### PR TITLE
Data layer refactored with TypeORM

### PR DESCRIPTION
**Pull request by intsco**
- Refactored 'src/modules/auth' and 'src/utils'

### CODE REVIEW

File: metaspace/graphql/utils.js [hunk 1]
```diff
@@ -132,11 +133,13 @@ const dbConfig = () => {
...
 
-let db = require('knex')({
-  client: 'pg',
-  connection: dbConfig(),
-  searchPath: ['engine', 'public']
-});
+const initDBConnection = (config = defaultDBConfig) => {
+  return Knex({
+    client: 'pg',
+    connection: config(),
```
[the remark this answers is not in the document]
Comment (intsco) [reply metaspace/graphql/utils.js]: I think it's just an old code I copied over. utils.js should be refactored one day too

File: metaspace/graphql/tests/imageUpload.test.js [hunk 2]
```diff
@@ -11,31 +11,44 @@ const chai = require('chai'),
...
 describe('imageUploadTest with fs and db backends', () => {
   ['db', 'fs'].forEach( (storageType) => {
 
     describe(`${storageType} storage type`, () => {
       let server;
+      let knex, knexAdmin;
 
       beforeAll(async () => {
-        server = await createImgServerAsync(config);
+        logger.info('> Before all');
```
[the remark this answers is not in the document]
Comment (intsco) [reply metaspace/graphql/tests/imageUpload.test.js]: That's very useful details

File: ansible/aws/aws_instance_manager.py [hunk 3]
```diff
@@ -205,8 +205,8 @@ def swap_prod_instance(self, suffix='beta'):
                                                  InstanceId=beta_inst.id)
         print('Associated Elastic IP: {}'.format(resp))
 
-        resp = prod_inst.stop()
-        print('Stopped old prod instance: {}'.format(resp))
+        # resp = prod_inst.stop()
+        # print('Stopped old prod instance: {}'.format(resp))
```
[the remark this answers is not in the document]
Comment (intsco) [reply ansible/aws/aws_instance_manager.py]: Yes, it's a part of a not yet finished feature.

File: metaspace/graphql/src/utils/db.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    // },
+    // debug: true
+  });
+};
+
+export interface DbRow {
+  id: number;
+}
+
+export const createUpdateTable = (knex: Knex) => {
```
[the remark this answers is not in the document]
Comment (intsco) [reply metaspace/graphql/src/utils/db.ts]: Yes, I thought about that, just wanted to reuse one function to update any table. But type safety is more important.

**Comment by intsco** [comment]
@LachlanStuart please take a look at the last commit when you have time

File: metaspace/graphql/src/modules/auth/model.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  googleId?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  emailVerificationToken?: string | null;
+
+  @Column({ type: 'timestamp without time zone', nullable: true, transformer: new MomentValueTransformer() })
+  emailVerificationTokenExpires?: Moment | null;
+
+  @Column({ type: 'boolean', default: false })
+  emailVerified?: boolean;
```
Comment (LachlanStuart) [inline metaspace/graphql/src/modules/auth/model.ts]: `emailVerified` shouldn't be optional as it's non-nullable in the database.

File: metaspace/graphql/src/modules/user/model.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  id: string;
+
+  @Column({ type: 'text' })
+  name: string;
+
+  @Column({ type: 'text'})
+  email: string;
+
+  @Column({ type: 'text', nullable: true })
+  role?: string;
```
Comment (LachlanStuart) [inline metaspace/graphql/src/modules/user/model.ts]: Can `role` be non-nullable, defaulting to 'user'?

**Comment by LachlanStuart** [comment]
Looks good :+1:

**Pull request merged**